### PR TITLE
:warning: remove kustomize/v1, go/v2 and go/v3, and configurations for Project Config v2 related to legacy Kubebuilder CLI version < 3

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
Remove deprecated plugins and doc sections as specific related tests.
- Remove golang v2 and v3
- Remove kustomize/v1 

**Motivation**
Maintainability and better usability 
More info: https://github.com/kubernetes-sigs/kubebuilder/discussions/3622
(began to prepare the master branch for a V4 release)

### CODE REVIEW

**Comment by camilamacedo86** [comment]
Hi @varshaprasad96 and @everettraven 

I will wait until next week for us start to removing all stuff 
So that we have an easy chance to release, like 3.14.1, and address any bugs or minor issues before.

Because changes in the API, it must be merged manually, so if you are able to look at and approve this one that would be great and we will only move forward anyway manually from next week

File: cmd/main.go [hunk 1]
```diff
@@ -43,24 +39,10 @@ func init() {
...
-		"Please, check the migration guide to learn how to upgrade your project"
-
-	// Bundle plugin which built the golang projects scaffold by Kubebuilder go/v3
-	gov3Bundle, _ := plugin.NewBundleWithOptions(plugin.WithName(golang.DefaultNameQualifier),
-		plugin.WithVersion(plugin.Version{Number: 3}),
-		plugin.WithDeprecationMessage(deprecateMessageGoV3Bundle),
-		plugin.WithPlugins(kustomizecommonv1.Plugin{}, golangv3.Plugin{}),
-	)
-
 	// Bundle plugin which built the golang projects scaffold by Kubebuilder go/v4 with kustomize alpha-v2
```
Comment (varshaprasad96) [inline cmd/main.go:42]: Nit
```suggestion
	// Bundle plugin which built the golang projects scaffold by Kubebuilder go/v4 with kustomize v2
```

File: pkg/config/store/yaml/store_test.go [hunk 2]
```diff
@@ -50,6 +51,8 @@ var _ = Describe("New", func() {
 var _ = Describe("yamlStore", func() {
 	const (
 		v2File = `version: "2"
```
Comment (varshaprasad96) [inline pkg/config/store/yaml/store_test.go:53]: Do we need this?
```suggestion
```

File: pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go [hunk 3]
```diff
@@ -79,10 +76,7 @@ func (s *apiScaffolder) InjectFS(fs machinery.Filesystem) {
 func (s *apiScaffolder) Scaffold() error {
 	log.Println("Writing scaffold for you to edit...")
 
-	//nolint: staticcheck
-	isGoV3 := plugin.IsLegacyLayout(s.config)
```
Comment (varshaprasad96) [inline pkg/plugins/golang/deploy-image/v1alpha1/scaffolds/api.go:83]: Glad to let these checks go :)

**Review by varshaprasad96 — COMMENTED** [review]
Took a first quick pass it looks good. I'll take a deeper look after the rebase.

**Comment by camilamacedo86** [comment]
HI @varshaprasad96 

Thank you a lot.
It is updated now.

Let's to do a patch release for what was fixed so far and then we can start to get this one merged and start to work in our MAJOR version bump Kubebuilder v4 as we have speaking so far where we have all cleaned up. 

:-)

**Comment by camilamacedo86** [comment]
I am moving within since
@varshaprasad96 already gave a first review
and we are just removing what is deprecated  (agreed and communicated a long time ago)

We need to merge this one in order to be able to add support for k8s 1.30 and go 1.22

**Pull request merged**